### ISSUE TITLE
Issues with 13" screen

### ISSUE THREAD

**Issue by blindside1** [safing/portmaster-ui#60]
**Checklist**:

- [x ] I'm using the official portmaster release (i.e no custom builds)
- [x ] I searched for similar/existing issues first.
- [ x] My issue is not mentioned in the Known Issues section of my [OS](https://github.com/safing/portmaster/wiki)

**What happened**: I installed portmaster on my 13" Dell XPS and the ui is too large and partially off the screen at the top so no way to correct it unless I resize from the bottom right. Also, there is no  -%PROGRAMDATA%\Portmaster\ļogs on my system

**What you expected to happen**: That it would automatically resize the ui to accomodate different sized screens

**How to reproduce it (as minimally and precisely as possible)**: Open the portmaster software

**Anything else we need to know?**: No

**Environment**: Win10 Pro

Portmaster Version: 0.4.3 alpha

<details>
  <summary>Versions from the `About` page in Portmaster's UI</summary>
  
</details>

Operating System:
- [ ] Windows 7
- [ ] Windows 8/8.1
- [x
<img width="960" alt="portmaster13bug" src="https://user-images.githubusercontent.com/48873350/81087186-ee500580-8ee8-11ea-99d3-c3dce9ded281.png">
 ] Windows 10
- [ ] Linux
  - Please provide the output of `cat /etc/os-release`

If applicable you can provide related sections from the log files and ensure to **remove sensitive or otherwise private information**.
 - Linux: `/var/lib/portmaster/logs`
 - Windows: `%PROGRAMDATA%\Portmaster\ļogs`

**Comment by ppacher** [comment]
Hi @blindside1,
thanks for reporting this.

I can confirm this also happens on linux. The portmaster app seems to always start with the same size. 

To confirm under linux:
```bash
# Start a nested X server with a screen limited to 800x600
Xephyr -screen 800x600 :2 &

# Launch a window manager and the portmaster app on that display (here xfce4 is used) 
export DISPLAY=:2
xfce4-session &
portmaster-control run app --data /var/lib/portmaster
```

The problematic line is https://github.com/safing/portmaster-ui/blob/develop/app/main.go#L92 . Seems like the webview library we are using does not support getting the maximum screen size so we'll need to find a cross-platform way to do that. 

For windows, we could use https://stackoverflow.com/a/48187712 as a starting point.

@dhaavi for this to be more user-friendly we should listen to resize events and store the preferred size in the configuration so we can restore it on the next start.

**Comment by dhaavi** [comment]
The UI Window now saves it's size. So you can resize it a size that fits you and it will reopen in that size.
Does that solve your problem?

**Comment by MPeti1** [comment]
I have the same problem with the latest version after a fresh install. How could I resize the window (lower the top of it) if the window header is out of screen?

Also, a bit unrelated but I think you should include the version number in the executable's metadata. Tried to find the version number there, but the executable says it's 1.0.0.0

**Comment by MPeti1** [comment]
Just after commenting I remembered that I can get the window header if I maximize Portmaster by pressing `Windows+Up`, and after that I was able to move the window so the header is now on-screen even if it's not maximized

When it was maximized, after closing and reopening, the window opened back not maximized, but like this:

![2020-10-27 00_34_42-Greenshot](https://user-images.githubusercontent.com/9215491/97239618-56bed600-17ec-11eb-89ff-21b001969955.png)

It seems it's not saved if the window was maximized last time, and also, that positioning and window size is not the same as before when it wasn't maximized

**Comment by dhaavi** [comment]
Thank you for providing additional input on this issue. We are currently in process of switching the component that controls the window and will keep this in mind. We will definitely look into this in the coming weeks.

> I have the same problem with the latest version after a fresh install. How could I resize the window (lower the top of it) if the window header is out of screen?

We currently not yet detect the current screen size, so smaller screens will have the window cut off.

> Also, a bit unrelated but I think you should include the version number in the executable's metadata. Tried to find the version number there, but the executable says it's 1.0.0.0

Version numbers in Windows executables are a bit tricky. I will look into adding integration for this into the build process. I have also opened a separate issue for this: https://github.com/safing/portmaster-ui/issues/87

> When it was maximized, after closing and reopening, the window opened back not maximized

With the current system we are not able to determine the _exact_ screen size, so opted for deducting a couple pixels in order to open up too big again. We will be looking into a clean solution for this.

> It seems it's not saved if the window was maximized last time, and also, that positioning and window size is not the same as before when it wasn't maximized

With the current system we cannot control window position.

**Comment by MPeti1** [comment]
Thank you for your response!

> With the current system we are not able to determine the exact screen size, so opted for deducting a couple pixels in order to open up too big again. We will be looking into a clean solution for this.

With maximization I wanted to mean the function that's accessible by this icon at the top right of windows:
![kép](https://user-images.githubusercontent.com/9215491/97509957-f1055200-1983-11eb-8ff3-0757693b2d2a.png)

Does that also need knowing the exact screen size? Saying this as a question, I'm not familiar with electron and similar apps from a developer perspective

**Comment by dhaavi** [comment]
This should now be fully resolved with [v0.6.0](https://github.com/safing/portmaster/releases/tag/v0.6.0). The Portmaster should notify you of the update.

If the problem persists with the new version, please reopen this issue.
